terraform/staging/main.tf

```diff
@@ -66,30 +66,121 @@ module "application" {
+}
+
+moved {
+  from = module.application.aws_iam_role.task_execution
+  to   = module.application_roles.aws_iam_role.task_execution
+}
+
+moved {
+  from = module.application.aws_iam_role_policy_attachment.task_execution_managed_policy
+  to   = module.application_roles.aws_iam_role_policy_attachment.task_execution_managed_policy
+}
+
+moved {
+  from = module.application.aws_iam_role_policy.parameter_access
+  to   = module.application_roles.aws_iam_role_policy.parameter_access
+}
+
+moved {
+  from = module.application.aws_iam_role_policy.secret_access
+  to   = module.application_roles.aws_iam_role_policy.secret_access
+}
+
+moved {
+  from = module.application.aws_iam_role.deployment
+  to   = module.application_roles.aws_iam_role.deployment
+}
+
+moved {
+  from = module.application.aws_iam_policy.allow_deployment
+  to   = module.application_roles.aws_iam_policy.allow_deployment
+}
+
+moved {
+  from = module.application.aws_iam_role_policy_attachment.allow_deployment
+  to   = module.application_roles.aws_iam_role_policy_attachment.allow_deployment
+}
+
+module "application_roles" {
+  source = "../modules/application_roles"
+
   github_actions_role_arn = "arn:aws:iam::815624722760:role/core-application-repo"
 
   prefix                               = local.prefix
   api_key_secret_arn                   = module.application_secrets.api_key_secret_arn
-  app_host                             = local.app_host
-  app_task_desired_count               = local.app_task_desired_count
-  application_port                     = local.application_port
   bulk_upload_bucket_access_policy_arn = module.bulk_upload.read_write_policy_arn
-  bulk_upload_bucket_details           = module.bulk_upload.details
   database_connection_string_arn       = module.database.rds_connection_string_arn
   database_data_access_policy_arn      = module.database.rds_data_access_policy_arn
-  ecs_security_group_id                = module.application_security_group.ecs_security_group_id
   export_bucket_access_policy_arn      = module.cds_export.read_write_policy_arn
-  export_bucket_details                = module.cds_export.details
   govuk_notify_api_key_secret_arn      = module.application_secrets.govuk_notify_api_key_secret_arn
-  load_balancer_target_group_arn       = module.front_door.load_balancer_target_group_arn
   os_data_key_secret_arn               = module.application_secrets.os_data_key_secret_arn
-  private_subnet_ids                   = module.networking.private_subnet_ids
-  rails_env                            = local.rails_env
   rails_master_key_secret_arn          = module.application_secrets.rails_master_key_secret_arn
-  redis_connection_string              = module.redis.redis_connection_string
   sentry_dsn_secret_arn                = module.application_secrets.sentry_dsn_secret_arn
-  sns_topic_arn                        = module.monitoring.sns_topic_arn
 }
 
 moved {

```

Comment:
Are these moved statements being left in for now because they haven't yet been applied to production? Presumably they've already been done for staging?

Reply:
Yep they should definitely remain for production as that won't be applied yet.

I could technically remove these from `staging/main.tf` now as I have applied it, but they're a useful record in case I need to move something back / elsewhere. My plan is to remove all these blocks at the end once the whole ticket is deemed OK 😀